### PR TITLE
Add Root.Security aspect and cleanup Smell.docs

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

File: coalib/bearlib/aspects/Security.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    class docs:
+        example = """
+        char buf[1024];
+        ssizet_t len;
+        if ((len = readlink("/modules/pass1", buf, sizeof(buf)-1)) != -1)
+            buf[len] = '\0';
+        """
+        example_language = 'C'
+        importance_reason = """
+        If, while writing data to a buffer we overwrite adjacent data or
```
Comment (adhikasp) [inline coalib/bearlib/aspects/Security.py]: IMO the docs seems too specific for a broad aspect like security.

**Comment by Asnelchristian** [comment]
@shreyans800755, ooh i see your point, but this example is not test sample; it is part of the ``Root.Security`` aspect documentation :sweat_smile: ....
Maybe this is not clear enough, look we want to keep ``aspects``well documented.... and that documentation consist of the definition, an example, the example language, an importance  reason, and fix suggestions.... You might wanna check other aspects like Root.smell found in coalib.bearlib.aspects.Smell or coala/aspect-docs repository

File: coalib/bearlib/aspects/Security.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        Security weaknesses can enable malicious users to bypass access controls
+        in order to obtain unauthorized privileges.
+        """
+        fix_suggestions = """
+        The only way to reduce the chance of a vulnerability being used against
+        a system is through constant vigilance, including careful system
+        maintenance (e.g. applying software patches), best practices in
+        deployment (e.g. the use of firewalls and access controls) and auditing
+        (both during development and throughout the deployment lifecycle).
+        """
```
Comment (userzimmermann) [inline coalib/bearlib/aspects/Security.py]: Can you really check such things as mentioned here with `coala`? ;)

**Comment by userzimmermann** [comment]
@Asnelchristian Forget about the second paragraph of https://github.com/coala/coala/pull/4360#pullrequestreview-45423505 ;) I just renamed this PR

File: coalib/bearlib/aspects/Security.py (hunk 2 above)
Comment (Asnelchristian) [inline coalib/bearlib/aspects/Security.py]: yeah... actually the bears available are not capable of all that :(

File: coalib/bearlib/aspects/Smell.py [hunk 3]
```diff
@@ -46,13 +58,12 @@ class docs:
...
+        maintainability of your source code.
         """
         fix_suggestions = """
-        When a class is wearing too many (functional) hats, think about
-        splitting it up:
+        When a class is wearing too many (functional) hats (too large
+        classes), you should probably think about splitting it up:
             * Extract class
             * Extract subclass
             * Extract interface
```
Comment (userzimmermann) [inline coalib/bearlib/aspects/Smell.py]: Lists should not be indented. And there should be a blank line above

File: coalib/bearlib/aspects/Smell.py [hunk 4]
```diff
@@ -6,25 +6,37 @@ class Smell:
...
-        * Feature envy
-        * Data clump
-        * Too large class
-        * Too long parameter list
-        etc...
+            * Feature envy
+            * Data clump
+            * Too large class
+            * Too long parameter list
+            * etc...
```
Comment (userzimmermann) [inline coalib/bearlib/aspects/Smell.py]: Lists should not be indented

File: coalib/bearlib/aspects/Smell.py [hunk 5]
```diff
@@ -6,25 +6,37 @@ class Smell:
...
+        """
+        fix_suggestions = """
+        There are several `refactoring techniques` that can be used to deal
+        with `code smells` including:
+            * Composing methods
+            * Movin features between objects
+            * Organizing data
+            * Simplifying conditional expressions
+            * Simplifying method calls
+            * Dealing with generalisation
```
Comment (userzimmermann) [inline coalib/bearlib/aspects/Smell.py]: Lists should not be indented. And there should be a blank line above

**Comment by userzimmermann** [comment]
@shreyans800755 Regarding https://github.com/coala/coala/pull/4360#pullrequestreview-44696846 : Examples are good and important in aspect docs to get some more intuitive understanding

**Comment by userzimmermann** [comment]
Let's just forget about https://github.com/coala/coala/pull/4360#pullrequestreview-45817907 ;P

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4360/commits/ba6b0713aee5b88b12b96863fe6876675370bc29 https://github.com/coala/coala/pull/4360/commits/fea06f52c99f62f6c76fecbafc0ed153f43706e0

**Comment by userzimmermann** [comment]
@rultor merge

**Pull request merged**